classes/migration/upgrade/v3_4_0/I9136_MigrateUniqueSiteId.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_4_0/I9136_MigrateUniqueSiteId.php
+ *
+ * Copyright (c) 2014-2023 Simon Fraser University
+ * Copyright (c) 2000-2023 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I9136_MigrateUniqueSiteId
+ *
+ * @brief Migrate UsageEvent plugin setting 'uniqueSiteId' to the site_settings table.
+ */
+
+namespace PKP\migration\upgrade\v3_4_0;
+
+use Illuminate\Support\Facades\DB;
+use PKP\core\PKPString;
+use PKP\install\DowngradeNotSupportedException;
+
+class I9136_MigrateUniqueSiteId extends \PKP\migration\Migration
+{
+    /**
+     * Run the migrations
+     */
+    public function up(): void
+    {
+        $newUniqueSiteId = DB::table('site_settings')
+            ->where('setting_name', '=', 'uniqueSiteId')
+            ->value('setting_value');
+        if (is_null($newUniqueSiteId)) {
+            $uniqueSiteId = DB::table('plugin_settings')
+                ->where('plugin_name', '=', 'usageeventplugin')
+                ->where('setting_name', '=', 'uniqueSiteId')
+                ->value('setting_value')
+                ??
+                PKPString::generateUUID();
+            DB::table('site_settings')->insert([
```

Comment:
I don't think it's going to fail, but an `updateOrInsert()` would be safer here.

Reply:
Hmmm... if there is already uniqueSiteId setting then it should not be overwritten, correct? So maybe it is better for the upgrade script to fail here?

Reply:
If it's falsy, I think it's ok to overwrite... But as this situation isn't going to happen anyway, anything is ok 🙂

Reply:
How we could check if it is falsy, what would be a falsy ID?

Reply:
If the value is empty/null. But this condition isn't going to happen here, so it's not needed to change anything :)

I just think the `updateOrInsert()` is the best choice when there are unique keys inside the entity, to avoid any kind of unexpected condition/failure.

Reply:
Thanks @jonasraoni. I will then leave it as it is for now...